scene/gui/control.cpp

```diff
@@ -680,18 +680,8 @@ Size2 Control::get_parent_area_size() const {
 
 // Positioning and sizing.
 
-Transform2D Control::_get_internal_transform() const {
-	Transform2D rot_scale;
-	rot_scale.set_rotation_and_scale(data.rotation, data.scale);
-	Transform2D offset;
-	offset.set_origin(-data.pivot_offset);
-
-	return offset.affine_inverse() * (rot_scale * offset);
-}
-
 void Control::_update_canvas_item_transform() {
-	Transform2D xform = _get_internal_transform();
-	xform[2] += get_position();
+	Transform2D xform = get_transform();
```

Comment:
This is the only potential difference in the behavior: `_get_internal_transform()` had no thread guard, both `get_position()` and `get_transform` have thread guard. Meaning in case of thread guard failing:
- Previously `xform` would end up with the result of `_get_internal_transform()` (`get_position()` would return `Vector2()` and hence no additional translation).
- Now `xform` would end up with `Transform2D()` as `get_transform()` would fail.

So in both cases `xform` would be incorrect (but differently). Should `_update_canvas_item_transform` have a thread guard on its own to abort further execution? :thinking: